src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/Presentation/ElisionBufferContent.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Media;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Projection;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.QuickInfo.Presentation
+{
+    /// <summary>
+    /// Creates quick info content out of the span of an existing snapshot.  The span will be
+    /// used to create an elision buffer out that will then be displayed in the quick info
+    /// window.
+    /// </summary>
+    internal class ElisionBufferContent
```

Comment:
Probably want to make this a ForegroundThreadAffinitizedObject.  And add asserts in our methods ot make sure the UI  stuff only happens on the UI thread.